### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by arybczak**

### CODE REVIEW

**Comment by arybczak** [comment]
@phadej  Why are the checks duplicated (pull_request & push)?

**Comment by phadej** [comment]
- `pull_request` is when you target PR into that branch.
- `push` is when you push into the branch

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#on

**Comment by phadej** [comment]
`haskell-ci` github actions backends lacks GHCJS support atm.

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# This GitHub workflow config has been generated by a script via
+#
+#   haskell-ci 'github' 'cabal.project'
```
Comment (phadej) [inline .github/workflows/haskell-ci.yml:3]: haskell-ci --config=cabal.haskell-ci cabal.project

**Comment by arybczak** [comment]
@phadej Ah, I see. I found https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/9 and applied a version of it to trigger builds only on pushes to master, but all PRs.

File: .github/workflows/haskell-ci.yml (hunk 1 above)
Comment (arybczak) [reply .github/workflows/haskell-ci.yml:3]: Oh. I see that also fixes the duplicate work problem :+1:

**Review by phadej — APPROVED** [review]
If this is unmodified workflow, then :+1;, let's switch.

I hope GHCJS support wont regress while I get to it on `haskell-ci` side.

**Pull request merged**